linter/lints/crl/lints.go

```diff
@@ -369,3 +377,176 @@ func hasMozReasonCodes(crl *crl_x509.RevocationList) *lint.LintResult {
+	if !tbs.SkipOptionalASN1(cryptobyte_asn1.INTEGER) {
+		return &lintFail
+	}
+
+	// Skip signature.
+	if !tbs.SkipASN1(cryptobyte_asn1.SEQUENCE) {
+		return &lintFail
+	}
+
+	// Skip issuer.
+	if !tbs.SkipASN1(cryptobyte_asn1.SEQUENCE) {
+		return &lintFail
+	}
+
+	// Read thisUpdate.
+	var thisUpdate cryptobyte.String
+	var thisUpdateTag cryptobyte_asn1.Tag
+	if !tbs.ReadAnyASN1Element(&thisUpdate, &thisUpdateTag) {
+		return &lint.LintResult{Status: lint.Error, Details: "Failed to read thisUpdate timestamp"}
+	}
+
+	// Lint thisUpdate.
+	err := lintTimestamp(&thisUpdate, thisUpdateTag)
+	if err != nil {
+		return &lint.LintResult{Status: lint.Error, Details: err.Error()}
+	}
+
+	// Peek (optional) nextUpdate.
+	if tbs.PeekASN1Tag(cryptobyte_asn1.UTCTime) || tbs.PeekASN1Tag(cryptobyte_asn1.GeneralizedTime) {
+
+		// Read nextUpdate.
+		var nextUpdate cryptobyte.String
+		var nextUpdateTag cryptobyte_asn1.Tag
+		if !tbs.ReadAnyASN1Element(&nextUpdate, &nextUpdateTag) {
+			return &lint.LintResult{Status: lint.Error, Details: "Failed to read nextUpdate timestamp"}
+		}
+
+		// Lint nextUpdate.
+		err = lintTimestamp(&nextUpdate, nextUpdateTag)
+		if err != nil {
+			return &lint.LintResult{Status: lint.Error, Details: err.Error()}
+		}
+	}
+
+	// Peek (optional) revokedCertificates.
+	if tbs.PeekASN1Tag(cryptobyte_asn1.SEQUENCE) {
+		// Read sequence of revokedCertificate.
+		var revokedSeq cryptobyte.String
+		if !tbs.ReadASN1(&revokedSeq, cryptobyte_asn1.SEQUENCE) {
+			return &lint.LintResult{Status: lint.Error, Details: "Failed to read revokedCertificate sequence"}
+		}
+
+		// Iterate over each revokedCertificate sequence.
+		for !revokedSeq.Empty() {
+			// Read revokedCertificate.
+			var certSeq cryptobyte.String
+			if !revokedSeq.ReadASN1Element(&certSeq, cryptobyte_asn1.SEQUENCE) {
+				return &lint.LintResult{Status: lint.Error, Details: "Failed to read revokedCertificate in sequence"}
+			}
+
+			if !certSeq.ReadASN1(&certSeq, cryptobyte_asn1.SEQUENCE) {
+				return &lint.LintResult{Status: lint.Error, Details: "Failed to read revokedCertificate in sequence"}
+			}
+
+			// Skip userCertificate (serial number).
+			if !certSeq.SkipASN1(cryptobyte_asn1.INTEGER) {
+				return &lintFail
+			}
+
+			// Read revocationDate.
+			var revocationDate cryptobyte.String
+			var revocationDateTag cryptobyte_asn1.Tag
+			if !certSeq.ReadAnyASN1Element(&revocationDate, &revocationDateTag) {
+				return &lintFail
+			}
+
+			// Lint revocationDate.
+			err = lintTimestamp(&revocationDate, revocationDateTag)
+			if err != nil {
+				return &lint.LintResult{Status: lint.Error, Details: err.Error()}
+			}
+		}
+	}
+	return &lint.LintResult{Status: lint.Pass}
+}
+
+func lintTimestamp(der *cryptobyte.String, tag cryptobyte_asn1.Tag) error {
+	// Preserve the original timestamp for length checking.
+	derBytes := *der
+	var tsBytes cryptobyte.String
+	if !derBytes.ReadASN1(&tsBytes, tag) {
+		return errors.New("x509: failed to read timestamp")
```

Comment:
Just realized that the other lint failure messages in this file don't have this sort of "x509:" or other source prefixing. I suspect that this error message stying is borrowed from the stdlib's `crypto/x509' library? I'd leave it out of these ones.

Reply:
You're right, I carried this over from the parsing func in crl_x509, I'll go ahead and remove these.